src/app/components/article.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { usePost } from '../store/hooks/usePost';
+import Thumbnail from './thumbnail';
+import Category from './category';
+import Author from './author';
+
+const Article = (props: any) => {
+    const article = usePost(props.postId);
+
+    return (
+        <article>
+            <div>
+                <Link to={`/${article.id}`}>
```

Comment:
Links need to use slugs not ids, for SEO benefit.

Reply:
You can then create a new hook to get posts by slugs, eg: `usePostBySlug( slug: string ): Post`